src/lib/utils.ts

```diff
@@ -2,3 +2,24 @@ export function randomIntFromInterval(min: number, max: number) {
 	// min and max included
 	return Math.floor(Math.random() * (max - min + 1) + min);
 }
+
+export function once<T extends Event>(fn?: (event: T) => void) {
+	return function (event: T) {
+		if (fn) fn(event);
+		fn = undefined;
+	};
+}
+
+export function preventDefault<T extends Event>(fn?: (event: T) => void) {
```

Comment:
also, is this change due to svelte 5? i haven't quite understood yet why we have to set it up this way for preventDefault and stopPropagation

Reply:
these 3 wrappers are straight from the svelte 5 docs. one regression with svelte 5 is that event modifiers had to move:
```
on:click|preventDefault={...}
```

all those special magic event directives are just regular attributes now.

here is the docs page, it explains it better: https://svelte-5-preview.vercel.app/docs/event-handlers

they say that it was just cool but actually an anti-pattern. those modifiers should be in the called function's code.

i wasn't sure which one to go with, that is why i used those wrapper functions; they mimick the old behavior. but it would be the same if we just called event.preventDefault at the top of the event handler functions, so maybe that  is better.

tho one advantage of the wrappers is that the event handler function does not need to explicitly write out the event parameter.